### DISCUSSION TITLE
Yarp.ReverseProxy.Forwarder.HttpForwarder: Information: Proxying to index.html

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/yarp#1763] [Q&A] [answered]
Hi,

I have my ASP NET webapp behind YARP, and with the following scenarios (with the URL)

- `https://comp2.net/app` without the trailing slash `/`, then the request is rerouted to `https://comp1.net/app/index.html`
- `https://comp2.net/app/` with a trailing slash `/`, then the webapp is hit with `https://comp1.net/app/`

The following is logged from YARP when rerouted to `index.html`

```Yarp.ReverseProxy.Forwarder.HttpForwarder: Information: Proxying to https://comp1.net/app/index.html``` 

If I run the same webapp without YARP, with the URL `https://comp1.net/app`, then I do not need to append the trailing slash `/` to make it work. However, if I do run with the URL `https://comp1.net/app/index.html`, then the webapp fails with 404, since the ASP NET webapp do not have an `index.html` to serve (it serves `Index`).

What is the proper approach here with YARP to make it work with ASP NET webapp? The webapp is created from the ASP NET template, so there's nothing special about the ASP NET webapp.

**Comment by samsp-msft** [comment, marked as answer]
Can you share your YARP configuration?
What client/browser are you using to access these URLs? Do you get the same behavior with cURL from the command line?

**Reply by ghost** [reply]
Hello!

Thanks for pointing me to client and cURL. Testing out with cURL first pointed me to the result that indeed that there were no `301` redirect from YARP when using the URL with or without the trailing `\`.

Going further, changing my Edge browser (the client) to Firefox produced the same result as cURL, there were no difference with or without trailing `\`.

Since I see this strange log in the Edge browser (301 from disk cache) given without the trailing `\` (and then the logging from YARP), I opened up the debugging window in Edge and disabled the cache from Edge. Clearing the Application data with Edge and then running the URL with or without the trailing `\` yields another result. *Now* the URL yields the same result as with cURL or Firefox.

I enabled the cache with Edge again, and now the `301 from disk cache` is gone. Now it works as expected, and YARP is happy.

Again, thanks for your reply!
